### PR TITLE
Improve tests for variables

### PR DESCRIPTION
**Pull request by joshuaogle**
Tests the output of variables more directly.

### CODE REVIEW

**Comment by joshuaogle** [comment]
The "has_value" matcher uses id's, so I'm going to leave those. For some reason I had to resort to "has_ruleset" to get it to match when I am mapping the psuedo-classes on to the lists, but I'd like to keep those as id's as well.

File: spec/bourbon/addons/buttons_spec.rb [hunk 1]
```diff
@@ -4,49 +4,50 @@
...
 
   context "expands plain buttons" do
     it "finds selectors" do
-      @buttons_list.each do |input|
-        expect(input).to have_rule("color: #ff0000")
-      end
+      list = @buttons_list.join(", ")
+      ruleset = "value: #{list};"
+
+      expect("#all-button-inputs").to have_ruleset(ruleset)
```
Comment (tysongach) [inline spec/bourbon/addons/buttons_spec.rb]: While this is still valid (via a fallback), we tweaked the name of this variable to simply `$all-buttons` in https://github.com/thoughtbot/bourbon/commit/a6374b6be375737cdfcb081d3a668b4865692038.

File: spec/fixtures/addons/buttons.scss [hunk 2]
```diff
@@ -1,17 +1,17 @@
 @import "setup";
 
-#{$all-button-inputs} {
-  color: #ff0000;
+#all-button-inputs {
```
Comment (tysongach) [inline spec/fixtures/addons/buttons.scss]: Can we just go ahead and use class selectors here and for all future tests. Better aligns with how we typically author and Hound will be a happy camper.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -15,8 +15,4 @@
   config.before(:all) do
     generate_css
   end
-
-  config.after(:all) do
-    clean_up
-  end
```
Comment (tysongach) [inline spec/spec_helper.rb]: What’s this removal all about?

File: spec/fixtures/addons/buttons.scss (hunk 2 above)
Comment (joshuaogle) [inline spec/fixtures/addons/buttons.scss]: I'd have to change the `has_value` matcher, but it should work. Sure

File: spec/spec_helper.rb (hunk 3 above)
Comment (joshuaogle) [inline spec/spec_helper.rb]: Doh. That was for testing. Glad you caught that.

File: spec/fixtures/addons/buttons.scss (hunk 2 above)
Comment (tysongach) [inline spec/fixtures/addons/buttons.scss]: Ah! I just saw your note below about that. No worries, then. For another time…

File: spec/fixtures/addons/text-inputs.scss [hunk 4]
```diff
@@ -1,17 +1,17 @@
 @import "setup";
 
-#{$all-text-inputs} {
-  color: #ff0000;
+.all-text-inputs {
+  value: $all-text-inputs;
```
Comment (tysongach) [inline spec/fixtures/addons/text-inputs.scss]: @joshuaogle Thoughts on disabling Hound on tests?

File: spec/support/matchers/have_value.rb [hunk 5]
```diff
@@ -1,15 +1,17 @@
 RSpec::Matchers.define :have_value do |expected|
   match do |variable|
-    selector_id = variable.sub("$", "#")
-    value_attribute = ParserSupport.parser.find_by_selector(selector_id)[0]
+    selector_id = variable.sub("$", ".")
```
Comment (tysongach) [inline spec/support/matchers/have_value.rb]: Should we rename `selector_id` to `selector_class` now?

File: spec/fixtures/addons/text-inputs.scss (hunk 4 above)
Comment (joshuaogle) [inline spec/fixtures/addons/text-inputs.scss]: :+1: I think there's perfectly valid times like this when bending the styleguide makes for better tests

File: spec/support/matchers/have_value.rb (hunk 5 above)
Comment (joshuaogle) [inline spec/support/matchers/have_value.rb]: Yep, fixed

File: spec/fixtures/addons/text-inputs.scss (hunk 4 above)
Comment (joshuaogle) [inline spec/fixtures/addons/text-inputs.scss]: On the other hand, we could probably use "content" instead of "value"

**Comment by tysongach** [comment]
This is looking good, @joshuaogle. Thanks for your hard work!

**Comment by tysongach** [comment]
@joshuaogle Is this ready to merge?

**Comment by joshuaogle** [comment]
@tysongach I forgot I still wanted to change those form `value: ` to `content: `. That's done and if you can give it another look over then it should be good to go.

**Comment by tysongach** [comment]
Ah yes, of course! It’s nice not seeing all the red now that `content` is being used. Great work!

LGTM! :+1:

**Pull request merged**